rest/config.go

```diff
@@ -1284,6 +1285,123 @@ func startServer(config *ServerConfig, sc *ServerContext) {
 	config.Serve(*config.Interface, CreatePublicHandler(sc))
 }
 
+func TestAuth(cluster *gocb.Cluster, username, password string, bucket string, requirePermissions ...string) (statusCode int, err error) {
+	_ = cluster
+	// TODO: Replace with cluster get Management endpoints
+	managementEndpoint := "http://work-couchbase.lan:8091"
+
+	// TODO: Get http client from cluster
+	httpClient := http.DefaultClient
+
+	body := strings.Join(requirePermissions, ",")
```

Comment:
Are you expecting bucket/collection-level permissions to already have that encoded in requirePermissions?  The use of bucket for roles but not permissions is a bit unclear.

Reply:
Yeah true... I should probably do that encoding like I am for roles